fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/LongArithmethicFunctionKeyExpression.java

```diff
@@ -129,15 +126,14 @@ public <S extends KeyExpressionVisitor.State, R> R expand(@Nonnull final KeyExpr
 
     @Nonnull
     @Override
-    public Value toValue(@Nonnull final CorrelationIdentifier baseAlias, @Nonnull final Type baseType, @Nonnull final List<String> fieldNamePrefix) {
-        ScalarTranslationVisitor scalarTranslationVisitor = new ScalarTranslationVisitor(arguments);
-        scalarTranslationVisitor.push(ScalarTranslationVisitor.ScalarVisitorState.of(baseAlias, baseType, fieldNamePrefix));
-        List<Value> argumentValues = new ArrayList<>(arguments.getColumnSize());
-        for (KeyExpression expression : arguments.normalizeKeyForPositions()) {
-            argumentValues.add(expression.expand(scalarTranslationVisitor));
-        }
-        BuiltInFunction<?> builtInFunction = FunctionCatalog.resolve(valueFunctionName, arguments.getColumnSize()).orElseThrow(() -> new RecordCoreArgumentException("unknown function", LogMessageKeys.FUNCTION, getName()));
-        return (Value) builtInFunction.encapsulate(argumentValues);
+    public Value toValue(@Nonnull final  List<? extends Value> argumentValues) {
+        Verify.verify(argumentValues.size() == arguments.getColumnSize());
+        final BuiltInFunction<?> builtInFunction =
+                FunctionCatalog.resolve(valueFunctionName,
+                                arguments.getColumnSize())
+                        .orElseThrow(() -> new RecordCoreArgumentException("unknown function",
+                                LogMessageKeys.FUNCTION, getName()));
+        return (Value)builtInFunction.encapsulate(argumentValues);
```

Comment:
I think this could profitably be extracted into a `protected` method on `FunctionKeyExpression` with `valueFunctionName` as an argument (along with `argumentValues`).

If the function does not have type overloading, that first argument might well be `getName()`. See https://github.com/FoundationDB/fdb-record-layer/pull/2899/commits/02185cf403edbba1e8d65b5802e77fe16980529a.